dirty_cat/datetime_encoder.py

```diff
@@ -0,0 +1,171 @@
+        Returns
+        -------
+        self
+            Fitted DatetimeEncoder instance.
+        """
+        self._validate_keywords()
+        self.to_extract_full = self.to_extract[:self.to_extract.index(self.extract_until) + 1]
+        self.to_extract_full.append("other")
+        if isinstance(X, pd.DataFrame):
+            self.colnames = X.columns
+        else:
+            self.colnames = None
+        X = check_input(X)
+        self.features_per_column = {}  # Features to extract for each column, after removing constant features
+        for i in range(X.shape[1]):
+            self.features_per_column[i] = []
+        # Check which columns are constant
+        for i in range(X.shape[1]):
+            for feature in self.to_extract_full:
+                if np.nanstd(self._extract_from_date(X[:, i], feature)) > 0:
+                    self.features_per_column[i].append(feature)
+            if self.add_day_of_the_week:
+                self.features_per_column[i].append("dayofweek")
+            if self.add_holidays:
+                self.features_per_column[i].append("holiday")
+
+        self.n_features_out_ = len(np.concatenate(list(self.features_per_column.values())))
+
+        return self
+
+    def transform(self, X, y=None):
+        """ Transform X by replacing each column with corresponding numerical features.
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, ) or (n_samples, 1)
+            The data to transform, where each column is a datetime feature.
+        Returns
+        -------
+        array, shape (n_samples, n_features_out_)
+            Transformed input.
+        """
+        X = check_input(X)
+        # Create a new dataframe with the extracted features, choosing only features that weren't constant during fit
+        X_ = np.empty((X.shape[0], self.n_features_out_), dtype=np.float64)
+        idx = 0
+        for i in range(X.shape[1]):
+            for j, feature in enumerate(self.features_per_column[i]):
+                X_[:, idx + j] = self._extract_from_date(X[:, i], feature)
+            idx += len(self.features_per_column[i])
+        return X_
+
+    def get_feature_names(self) -> List[str]:
+        """
+        Returns clean feature names with format "<column_name>_<new_feature>"
+        if the original data has column names, otherwise with format
+        "<column_index>_<new_feature>". new_feature is one of ["year", "month",
+        "day", "hour", "minute", "second", "millisecond", "microsecond",
+        "nanosecond", "dayofweek", "holiday"]
+        """
+        feature_names = []
+        for i in self.features_per_column.keys():
+            prefix = str(i) if self.colnames is None else self.colnames[i]
+            for feature in self.features_per_column[i]:
+                feature_names.append(f"{prefix}_{feature}")
+        return feature_names
+
+    def get_feature_names_out(self, input_features=None) -> List[str]:
```

Comment:
Can you invert get_feature_names and get_feature_names_out: get_feature_names will be removed in a little while, when we no longer depend on old versions of scikit-learn.